spec/models/tta_activity_report_spec.rb

```diff
@@ -0,0 +1,77 @@
+require "rails_helper"
+
+RSpec.describe TtaActivityReport, type: :model do
+  let(:access_token) { HsesAccessToken.new }
+  let(:display_id) { "R14-AR-200" }
+
+  subject do
+    # models will retrieve the tta_report_id from FakeData based on the display ID
+    described_class.new display_id, access_token
+  end
+
+  describe "validations" do
+    it "is valid" do
+      expect(subject).to be_valid
+    end
+
+    context "unauthorized report" do
+      let(:display_id) { "R14-AR-403" }
+
+      it "is invalid" do
+        expect(subject).to_not be_valid
+      end
+    end
+  end
+
+  describe "#ttahub_url" do
+    it "returns the URL to view the report in TTA Hub" do
+      expect(subject.ttahub_url).to match %r{https://ttahub.ohs.acf.hhs.gov/activity-reports/view/\d+}
+    end
+
+    it "raises an error when there is an error with the api" do
+      # This displa ID will trigger a 403 error in the TTA Hub API
+      subject.display_id = "R14-AR-403"
+      expect { subject.ttahub_url }.to raise_error(Api::Error)
+    end
+  end
+
+  describe "#topics" do
+    it "returns an array of topics" do
```

Comment:
[suggestion] should probably have at least one of each type of field that tests that the return is `nil` if the path or the attribute itself is not present.